tests/model/ArrayListTest.php

```diff
@@ -426,6 +426,25 @@ public function testMultiSort() {
 	}
 
 	/**
+	 * Check that we don't cause recursion errors with array_multisort() and customised ViewableData objects
+	 */
+	public function testSortWithRecursiveReferences() {
+		$items = ArrayList::create();
+
+		// Objects need the same sort value so that their properties are compared by array_multisort
+		$items->add(DataObject::create(array('Order' => 1)));
+		$items->add(DataObject::create(array('Order' => 1)));
+
+		// Customising the object adds recursive dependencies (between the object and its customised instance)
+		foreach ($items as $item) {
+			$item->customise(array());
```

Comment:
We are tightly coupling to customise's behaviour here - we probably shouldn't do that.

Isn't this patch solving a problem with recursive references, not customise?

I see no reason not to solve this for all objects rather than just `ViewableData`